File: testprojects/java/benchmark/infrastructure/database/derby/DerbyConnectionSettings.java
package infrastructure.database.derby;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

import infrastructure.database.Datasource;
//Functional requirement 2.3
//Test case 3: Only the classes in package infrastructure.socialmedia (including all subpackages and subclasses) are allowed to use the JDBC library
//Result: FALSE
public abstract class DerbyConnectionSettings extends Datasource{
	private static final String DERBY_DB_CONN = "jdbc:derby://localhost:1527/myDB;create=true;";
	private static final String DERBY_DB_USER = "me";
	private static final String DERBY_DB_PASS = "password";
	
	public Connection getConnection() {
		Connection connection = null;
		try {
			connection = DriverManager.getConnection(DERBY_DB_CONN, DERBY_DB_USER, DERBY_DB_PASS);
		} catch (SQLException ex) {
			System.out.println(ex);
		}
		return connection;
	}
}

File: testprojects/c#/benchmark/infrastructure/database/derby/DerbyConnectionSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data.SQLite;

namespace CSharpBenchmark.infrastructure.database.derby
{
    //Functional requirement 2.3
    //Test case 3: Only the classes in package infrastructure.socialmedia (including all subpackages and subclasses) are allowed to use the SQLite library
    //Result: FALSE
    public abstract class DerbyConnectionSettings : Datasource
    {
        private static readonly String CONN_STRING = "Data Source=DemoT.db;Version=3;New=False;Compress=True";
        SQLiteConnection connection = new SQLiteConnection(CONN_STRING);

        public override SQLiteConnection getConnection()
        {
            connection.Open();
            return connection;
        }
    }
}